src/index.ts

```diff
@@ -81,14 +81,28 @@ export async function createVerifiableCredentialJwt(
  *
  * @param payload `PresentationPayload` or `JwtPresentationPayload`
  * @param holder `Issuer` of the Presentation JWT (holder of the VC), signer and algorithm that will sign the token
+ * @param options `CreatePresentationOptions` allows to pass additional values to the resulting JWT payload
  * @return a `Promise` that resolves to the JWT encoded verifiable presentation or rejects with `TypeError` if the
  * `payload` is not W3C compliant
  */
 export async function createVerifiablePresentationJwt(
   payload: JwtPresentationPayload | PresentationPayload,
-  holder: Issuer
+  holder: Issuer,
+  options: CreatePresentationOptions = {}
 ): Promise<JWT> {
   const parsedPayload: JwtPresentationPayload = { iat: undefined, ...transformPresentationInput(payload) }
+
+  // add challenge to nonce
+  if (options.challenge) {
+    parsedPayload.nonce = options.challenge
```

Comment:
if `nonce` is already present in the payload, perhaps this should throw an error.

Reply:
agreed, I can add this.

Reply:
Updated to only overwrite if `nonce` is not already present (same way it's done in `transformPresentationInput`).